pkg/service/replay/replay.go

```diff
@@ -951,6 +970,170 @@ func (r *Replayer) RunApplication(ctx context.Context, appID uint64, opts models
 	return r.instrumentation.Run(ctx, appID, opts)
 }
 
+func (r *Replayer) GetTestSetConf(ctx context.Context, testSet string) (*models.TestSet, error) {
+	return r.TestSetConf.Read(ctx, testSet)
+}
+
+func (r *Replayer) Templatize(ctx context.Context, testSets []string) error {
+	if len(testSets) == 0 {
+		if len(r.config.Templatize.TestSets) == 0 {
+			allTestSets, err := r.GetAllTestSetIDs(ctx)
+			if err != nil {
+				r.logger.Error("failed to get the test sets", zap.Error(err))
```

Comment:
remove this else block and return if there is an error, you should use `utils.LogError` rather than the `r.logger.Error`